lib/ukcpio/cpio.c

```diff
@@ -58,6 +58,37 @@ int uk_syscall_r_utime(const char *, const struct utimbuf *);
 int uk_syscall_r_mkdir(const char *, mode_t);
 int uk_syscall_r_symlink(const char *, const char *);
 int uk_syscall_r_stat(const char *, struct stat *);
+int uk_syscall_r_unlinkat(int, const char *, int);
+int uk_syscall_r_rename(const char *, const char *);
+
+static int
+try_remove(const char *path)
+{
+	int r;
+
+	uk_pr_info("Trying to unlink %s\n", path);
+	r = uk_syscall_r_unlinkat(AT_FDCWD, path, 0);
+	if (!r || r == -ENOENT)
+		return 0;
+
+	uk_pr_info("Trying to rmdir %s\n", path);
+	r = uk_syscall_r_unlinkat(AT_FDCWD, path, AT_REMOVEDIR);
+	if (!r || r == -ENOENT) {
+		return 0;
+	} else {
+		char newpath[PATH_MAX];
+		char *newend = newpath + strlcpy(newpath, path, PATH_MAX);
+
+		if (newend - newpath + 2 > PATH_MAX) {
+			uk_pr_err("Cannot rename %s, path too long\n", path);
+			return -ENAMETOOLONG;
+		}
+		strcpy(newend, ".0");
```

Comment:
I would rather we did
 ```C
 newend[0] = '.';
 newend[1] = '0';
 newend[2] = '\0';
 ```
instead of a function call.

Is there a reason why `strcpy` was used or just a minor oversight.

Reply:
> Is there a reason why strcpy was used or just a minor oversight.
Largely to keep the appended string configurable/of arbitrary length. I doubt you'd argue the same point if the line had been
```c
strcpy(newend, ".cpio_rename~")
```
There's nothing fundamental about `.0`, so I don't think it makes sense to special-case it this hard.